components/post_markdown/post_markdown.test.tsx

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+import {Posts} from 'mattermost-redux/constants';
+import {Post, PostType} from 'mattermost-redux/types/posts';
+
+import PostMarkdown from 'components/post_markdown/post_markdown';
+import Markdown from 'components/markdown';
+
+describe('components/PostMarkdown', () => {
+    const baseProps = {
+        imageProps: {},
+        isRHS: false,
+        message: 'message',
+        mentionKeys: [{key: 'a'}, {key: 'b'}, {key: 'c'}],
+        channelId: 'channel-id',
+    };
+
+    test('should not error when rendering without a post', () => {
+        const props = {...baseProps};
+        Reflect.deleteProperty(props, 'post');
+
+        shallow(<PostMarkdown {...props}/>);
+    });
+
+    test('should render properly with an empty post', () => {
+        const wrapper = shallow(
+            <PostMarkdown {...baseProps}/>,
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should render properly with a post', () => {
+        const props = {
+            ...baseProps,
+            message: 'See ~test',
+            post: {
+                id: '123',
```

Comment:
If you want to reduce some code bloat, we have test helpers to mock out objects like `Post` in `utils/test_helper`